File: Services/TaskService.cs
using Microsoft.EntityFrameworkCore;
using TaskManagerAPI.Data;
using TaskManagerAPI.Models.DTO;
using TaskManagerAPI.Models.Entities;

namespace TaskManagerAPI.Services;

public interface ITaskService
{
    Task<IEnumerable<UserTask>> GetAllTasks();
    Task CreateTask(UserTaskDto model);
    Task UpdateTask(int id, UpdateUserTaskDto model);
    Task DeleteTask(int id);
}

public class TaskService : ITaskService
{
    private TaskManagerContext _context;

    public TaskService(TaskManagerContext context)
    {
        _context = context;
    }

    public async Task<IEnumerable<UserTask>> GetAllTasks()
    {
        return await _context.Tasks.ToListAsync();
    }

    public async Task CreateTask(UserTaskDto model)
    {
        var task = new UserTask
        {
            Title = model.Title,
            Description = model.Description,
            DueDate = model.DueDate,
            CreatedDate = DateTime.Now,
            IsCompleted = false,
        };

        _context.Tasks.Add(task);
        await _context.SaveChangesAsync();
    }

    public async Task UpdateTask(int id, UpdateUserTaskDto model)
    {
        var task = _context.Tasks.FirstOrDefault(t => t.Id == id) ?? throw new KeyNotFoundException("That task doesnt exist.");

        task.Title = model.Title;
        task.Description = model.Description;
        task.DueDate = model.DueDate;
        task.IsCompleted = model.IsCompleted;

        await _context.SaveChangesAsync();
    }

    public async Task DeleteTask(int id)
    {
        var task = await _context.Tasks.FirstOrDefaultAsync(t => t.Id == id) ?? throw new KeyNotFoundException("That task doesnt exist.");

        _context.Tasks.Remove(task);
        _context.SaveChanges();
    }
}

File: View/task-manager/src/Services/TaskService.ts
import axios from "axios";

export async function getAllTasks() {
  try {
    const response = await axios.get(
      "https://localhost:7258/api/Tasks/Get-all-tasks"
    );
    return response.data;
  } catch (error) {
    console.log(error);
    return [];
  }
}

export async function createTask(taskData: object) {
  try {
    const response = await axios.post(`https://localhost:7258/api/Tasks/Create`, taskData);
    return response.data;
  } catch (error) {
    console.log(error);
    return error;
  }
}

export async function updateTask(
  id: number,
  title: string,
  description: string,
  dueDate: Date,
  isCompleted: boolean
) {
  try {
    const response = await axios.put(
      `https://localhost:7258/api/Tasks/Update/${id}`,
      {
        title,
        description,
        dueDate,
        isCompleted,
      }
    );
    return response.data;
  } catch (error) {
    console.log(error);
    return error;
  }
}

export async function deleteTask(id: number) {
  try {
    const response = await axios.delete(
      `https://localhost:7258/api/Tasks/Delete${id}`
    );
    return response.data;
  } catch (error) {
    console.log(error);
    return error;
  }
}
